fraud/proof.go

```diff
@@ -39,12 +39,21 @@ func (p ProofType) String() string {
 	}
 }
 
+func toProof(p int32) ProofType {
+	switch p {
+	case 0:
+		return BadEncoding
+	default:
+		panic(fmt.Sprintf("fraud: invalid proof type: %d", p))
```

Comment:
Wouldn't this allow someone to halt others' nodes by responding with a fake proof with any index other than 0?

Reply:
Yes, this makes sense. But from the other side, we can skip some Fraud Proof that is not supported

Reply:
Our node could have older version(that does not support some types of fraud proofs), so will can add valid peer to a black list

Reply:
So just for clarification: The reason this is okay is because the node will just restart, and won't get this BEFP anymore because it has blocklisted the peer? But what about in the case of a Sybil attack from a malicious proof spammer?

Reply:
Sorry, did not get your questions